crates/rattler_solve/src/libsolv/input.rs

```diff
@@ -0,0 +1,258 @@
+//! Contains business logic that loads information into libsolv in order to solve a conda
+//! environment
+
+use crate::libsolv::util;
+use crate::libsolv::util::PackageExtension;
+use crate::libsolv::wrapper::keys::*;
+use crate::libsolv::wrapper::pool::Pool;
+use crate::libsolv::wrapper::repo::Repo;
+use crate::libsolv::wrapper::repodata::Repodata;
+use crate::libsolv::wrapper::solvable::SolvableId;
+use rattler_conda_types::{GenericVirtualPackage, RepoDataRecord};
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::ffi::{CString, NulError};
+
+/// Adds [`RepoDataRecord`] to this instance
+pub fn add_repodata_records(
+    pool: &Pool,
+    repo: &Repo,
```

Comment:
I think its important that the passed in `Repo` was created from the `Pool` right? Not doing so might result in undefined behavior. Would it be possible to get a reference to pool from the repo? Or can you add an assertion to ensure this is the case?

Reply:
Why do you think that passing a `Repo` that does not belong to the `Pool` will result in UB? At least on our side, when we resolve `SolvableId`s we make sure to avoid UB. Are there some assumptions in the libsolv side that we should take into account too? I haven't checked libsolv's code, and maybe it pays off to be extra careful here anyway, so I'll add the check just in case (even without UB it will result in more predictable errors if anyone ever misuses the function)

Reply:
Because a `Repo` references data from the `Pool` from which it was created. The current API allows this:

```rust
let pool_a = Pool::new();
let pool_b = Pool::new();
let repo = pool_a.create_repo("cool");
add_repodata_records(pool_b, repo); // <-- could result in undefined behavior.
```

The repo also contains a pointer to the pool. If we just add an assert that that pointer matches the pass in pool we can be sure that its correct.